crates/epaint/src/stroke.rs

```diff
@@ -52,3 +54,116 @@ impl std::hash::Hash for Stroke {
         color.hash(state);
     }
 }
+
+/// How paths will be colored.
+#[derive(Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub enum ColorMode {
+    /// The entire path is one solid color, this is the default.
+    Solid(Color32),
+
+    /// Provide a callback which takes in a UV coordinate and converts it to a color. The UVs are relative to the bounding box of the path.
+    /// If you need something more custom, consider [`ColorMode::UVBounds`]
+    ///
+    /// **This cannot be serialized**
+    #[cfg_attr(feature = "serde", serde(skip))]
+    UV(Arc<Box<dyn Fn(Pos2) -> Color32 + Send + Sync>>),
```

Comment:
This needs a bit more explanation. Is `Pos2(0, 0)` the top right of the bounding box and `Pos2(1,1,)` the bottom right of it?